app/importers/file_importers/base_csv_importer.rb

```diff
@@ -0,0 +1,37 @@
+class BaseCsvImporter < Struct.new :school_scope, :client, :log, :progress_bar, :import_record_detail
+  def import
+    import_record_detail.start
+    before_import if self.respond_to?(:before_import)
+
+    @data = CsvDownloader.new(
+      log: log, remote_file_name: remote_file_name, client: client, transformer: data_transformer
+    ).get_data
+
+    @data.each.each_with_index do |row, index|
+      if filter.include?(row)
+        import_record_detail.log_processed
+        import_row(row)
+      else
+        import_record_detail.log_excluded
+      end
+      ProgressBar.new(log, remote_file_name, @data.size, index + 1).print if progress_bar
+    end
+    after_import if self.respond_to?(:after_import)
```

Comment:
Hey @jhilde! I definitely see where you're coming from with the `BaseCsvImporter` class. It feels tidy and clean to wrap up shared logic into one place! Check out #1147 though, we used to have all this logic in one place called `FileImport`, but then we decided to duplicate it across all the importers. 

Why? We wanted to avoid splitting up flow-of-control code across multiple classes. Checks like `after_import if self.respond_to?(:after_import)` make it hard to read through the flow-of-control for each importer class, because you have to skip around in the code to discover what the import process looks like. 

Especially when we have patterns like "import rows, then delete some rows", we want it to be really discoverable and clear at the top of each class how the flow of control pattern works.

Reply:
Yes I completely appreciated the removal of `FileImport'! For me it was mostly because it used a bit of convoluted composition rather than inheritance. 

My theory for using inheritance here is:
1) Lots of duplicate code. 
2) A desire to push toward increasing configuration over code (a common theme with me) as we anticipate expanding our base of districts.
3) Creating a base class that is thoroughly tested rather than expect each importer to repeat the same test with different fields.

May not be the time which is fine...just got really tired of copying and pasting. ;-)